azure/scope/managedcontrolplane.go

```diff
@@ -621,9 +621,54 @@ func (s *ManagedControlPlaneScope) ManagedClusterSpec() azure.ASOResourceSpecGet
 		}
 	}
 
+	if s.ControlPlane.Spec.SecurityProfile != nil {
+		managedClusterSpec.SecurityProfile = s.GetManagedClusterSecurityProfile()
+	}
+
 	return &managedClusterSpec
 }
 
+// GetManagedClusterSecurityProfile gets the security profile for managed cluster.
+func (s *ManagedControlPlaneScope) GetManagedClusterSecurityProfile() *managedclusters.ManagedClusterSecurityProfile {
```

Comment:
nit, since we don't need this outside any other package:
```suggestion
func (s *ManagedControlPlaneScope) getManagedClusterSecurityProfile() *managedclusters.ManagedClusterSecurityProfile {
```